ACPs/75_consensus_proofs/75-consensus-proofs.md

```diff
@@ -0,0 +1,87 @@
+
+One key invariant is that nodes cannot verify a block that was produced at a P-Chain height that they have not heard of yet. Verifiers assume that a block height that is beyond the P-Chain height that they have accepted is invalid because that block does not exist yet from the verifier’s perspective.
+
+If many nodes disagree about the current tip of the P-Chain, it can lead to a liveness failure where a subnet is not able to produce any blocks because nodes might have copies of the P-Chain that are out-of-sync with each other. In practice, this almost never happens because nodes produce blocks with a P-Chain height in the past, using a few blocks as a buffer since it’s likely that most nodes would have accepted an old block. This however, relies on an assumption that validators are constantly making progress in consensus to prevent the subnet from potentially stalling. This leaves an open concern where the P-Chain stalling on a node would prevent it from verifying any blocks, leading to a subnet potentially unable to produce blocks if many validators stalled at different heights due to a P-Chain outage.
+
+If ProposerVM blocks were produced with a proof of the P-Chain block to be verified against instead of just its height, verifiers could continue making progress by requesting any block they had not accepted yet from another peer and verifying their corresponding acceptance proofs. If a block’s proof is valid, the blocks can be executed in-order locally to finally verify the proposed subnet block.
+
+Figure 1: A Validator is verifying a subnet’s block `Z` which references an unknown P-Chain block `C` in its block header
+
+![figure 1](./1.jpg)
+
+Figure 2: A Validator requests the blocks and proofs for `B` and `C` from a peer
+
+![figure 2](./2.jpg)
+
+Figure 3: The Validator accepts the P-Chain blocks and is now able to verify `Z`
+
+![figure 3](./3.jpg)
+
+## Specification
+
+Note: The following is pseudocode and is provided only as visual aid.
+
+### ProposerVM
+
+The block header will require an update to include a Proof of the parent block
+to be verified at corresponding to the existing height field.
+
+```diff
+type BlockHeader struct {
+    PChainHeight uint64
+    Timestamp    time.Time
+    Proposer     ids.NodeID
++   Proof        []byte
+  }
+```
+
+### P2P
+
+The `Ancestors` message is replaced with the `containers`
+field which will supply a set of containers with their corresponding proofs.
+
+```diff
+message Ancestors {
+    bytes chain_id = 1;
+    uint32 request_id = 2;
+-   repeated bytes containers = 3;
++   repeated Container containers = 4;
+}
+```
+
+The `Container` message is used to send a container with its corresponding
+consensus proof.
+
+```diff
++ message Container {
++   bytes container = 1;
++   bytes proof = 2;
```

Comment:
Do we want to be more specific about what we think a `proof` could be here (i.e. a BLS Multi-Signature)? Otherwise, I think "Consensus Proof" could be interpreted a lot of different ways.